services/datamanager/builtin/sync/file_deletion.go

```diff
@@ -0,0 +1,260 @@
+package sync
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/benbjohnson/clock"
+	"github.com/pkg/errors"
+
+	"go.viam.com/rdk/data"
+	"go.viam.com/rdk/logging"
+	"go.viam.com/rdk/utils/diskusage"
+)
+
+var (
+	// FSThresholdToTriggerDeletion temporarily public for tests.
+	FSThresholdToTriggerDeletion = .90
+	// CaptureDirToFSUsageRatio temporarily public for tests.
+	CaptureDirToFSUsageRatio = .5
+)
+
+var errAtSizeThreshold = errors.New("capture directory has reached or exceeded disk usage threshold for deletion")
+
+func deleteExcessFilesOnSchedule(
+	ctx context.Context,
+	fileTracker *fileTracker,
+	captureDir string,
+	deleteEveryNth int,
+	clock clock.Clock,
+	logger logging.Logger,
+) {
+	if runtime.GOOS == "android" {
+		logger.Debug("file deletion if disk is full is not currently supported on Android")
+		return
+	}
+	t := clock.Ticker(CheckDeleteExcessFilesInterval)
+	defer t.Stop()
+	for {
+		if err := ctx.Err(); err != nil {
+			return
+		}
+
+		select {
+		case <-ctx.Done():
+			return
+		case <-t.C:
+			maybeDeleteExcessFiles(ctx, fileTracker, captureDir, deleteEveryNth, clock, logger)
+		}
+	}
+}
+
+func maybeDeleteExcessFiles(
+	ctx context.Context,
+	fileTracker *fileTracker,
+	captureDir string,
+	deleteEveryNth int,
+	clock clock.Clock,
+	logger logging.Logger,
+) {
+	start := clock.Now()
+	logger.Debug("checking disk usage")
+	usage, err := diskusage.Statfs(captureDir)
+	logger.Debugf("disk usage: %s", usage)
+	if err != nil {
+		logger.Error(errors.Wrap(err, "error checking file system stats"))
+		return
+	}
+
+	if usage.SizeBytes == 0 {
+		logger.Error("captureDir partition has size zero")
+		return
+	}
+	deletedFileCount, err := deleteExcessFiles(
+		ctx,
+		fileTracker,
+		usage,
+		captureDir,
+		deleteEveryNth,
+		logger)
+
+	duration := clock.Since(start)
+
+	switch {
+	case err != nil:
+		logger.Errorw("error deleting cached datacapture files", "error", err, "execution time", duration.Seconds())
+	case deletedFileCount > 0:
+		logger.Infof("%d files have been deleted to avoid the disk filling up, execution time: %f", deletedFileCount, duration.Seconds())
+	default:
+		logger.Infof("no files delete, execution time: %s", duration)
```

Comment:
```suggestion
		logger.Infof("no files deleted, execution time: %f", duration.Seconds())
```

Reply:
I've updated `delete` -> `deleted` but don't want to change seconds to a float as I think `200 ms` is easier to read than `0.2 s`

Reply:
oh i just suggested that for consistency, your other log lines in this block are using `duration.Seconds()`

Reply:
thanks that is fixed now too